app/models/colyseus-models/status.ts

```diff
@@ -1074,4 +1083,30 @@ export default class Status extends Schema implements IStatus {
       this.pokerusCooldown -= dt
     }
   }
+
+  triggerLocked(duration: number, pkm: PokemonEntity) {
+    if (
+      !this.locked && // lock cannot be stacked
+      !this.runeProtect
+
+    ) {
+
+      if (pkm.status.enraged) {
+        duration = duration / 2
+      }
+
+      this.locked = true
+      this.lockedCooldown = Math.round(duration)
+      pkm.range = 1
```

Comment:
should change pokemon state to IdleState

Reply:
Correct me if I'm wrong. Would changing Pokemon state to IdleState make Locked functionally the same as Freeze, as the Pokemon would be unable to attack?
I've addressed all the other comments in my local repo except this one, just wanted to check before committing and pushing this

Reply:
right, i see we don't have a IdleState > Attacking state condition. It's probably a mistake because transitionning to moving state introduce a ~500ms delay before attacking at range. But maybe it's intentional ? Anyway, idle state is almost not used during battle so not a big deal. My intent with this remark was to not have a locked pokemon in MovingState since it obviously can't move. It avoids checking against a canMove condition at every update. But if its simpler to handle this logic in Movingstate, I can bear with that